lib/mercury/cps.rb

```diff
@@ -35,18 +35,18 @@ def run(*args, &k)
     #   Cps and returns a Cps
     def and_then(&pm)
       Cps.new do |*args, &k|
-        self.run(*args) do |*args2|
-          next_cps = pm.call(*args2)
-          next_cps.is_a?(Cps) or raise "'and_then' block did not return a Cps object. Did you want 'and_lift'? at #{pm.source_location}"
+        run(*args) do |*args2|
+          next_cps = yield(*args2)
```

Comment:
How does `yield` make the code more readable, when the block is accepted explicitly as `pm` and the next line refers to `pm`?

When multiple blocks are in scope, I find it easier to keep track of them when they're all used explicitly.